spring_first1/src/main/java/com/example/demo/controller/LogoutController.java

```diff
@@ -0,0 +1,44 @@
+package com.example.demo.controller;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.example.demo.controller.form.LoginForm;
+
+
+
+@Controller
+
+@RequestMapping("/logout")
+public class LogoutController {
+	@GetMapping
+	String getLogout(HttpServletRequest request, @ModelAttribute LoginForm logiForm) {
+		HttpSession session = request.getSession(false);
+
+		if (session != null){
+			session.invalidate();
```

Comment:
```suggestion
			session.removeAttribute("login_id");
```
invalidate()だと完全にセッション破棄してくれない場合があるので
removeAttributeでセッションに詰めた値を消した方が良いです。
セッションにidがあればログイン中というルールを作ってしまえば
消すのはidだけで良いので。